crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,479 @@
+            }
+            _ => false,
+        };
+    }
+
+    false
+}
+
+/// Determines whether the operator is one of those used in range assertions (< or <=)
+fn is_range_assertion_operator(operator: Option<JsBinaryOperator>) -> bool {
+    matches!(
+        operator,
+        Some(JsBinaryOperator::LessThan | JsBinaryOperator::LessThanOrEqual)
+    )
+}
+
+/// Determines whether the nodes compose an inside range assertion, like `0 <= x && x < 1`
+fn is_inside_range_assertion(
+    operator: &JsLogicalOperator,
+    left_binary_expression: &JsBinaryExpression,
+    right_binary_expression: &JsBinaryExpression,
+) -> bool {
+    let left_identifier = left_binary_expression.right().ok();
+    let right_identifier = right_binary_expression.left().ok();
+
+    if !matches!(operator, JsLogicalOperator::LogicalAnd)
+        || !is_same_identifier(&left_identifier, &right_identifier)
+    {
+        return false;
+    }
+
+    let left_literal = extract_string_value(left_binary_expression.left().as_ref().ok());
+    let right_literal = extract_string_value(right_binary_expression.right().as_ref().ok());
+
+    match (left_literal, right_literal) {
+        (None, None) => false,
+        (Some(_), None) => true,
+        (None, Some(_)) => true,
+        (Some(left_value), Some(right_value)) => {
+            compare_literals(left_value.as_str(), right_value.as_str())
+        }
+    }
+}
+
+/// Compare literals as number if possible, compare as string otherwise
+fn compare_literals(left_string_value: &str, right_string_value: &str) -> bool {
+    match (
+        left_string_value.parse::<f64>(),
+        right_string_value.parse::<f64>(),
+    ) {
+        (Ok(left_number_value), Ok(right_number_value)) => left_number_value <= right_number_value,
+        _ => left_string_value <= right_string_value,
+    }
+}
+
+/// Determines whether the nodes compose an outside range assertion, like `x < 0 || 1 <= x`
+fn is_outside_range_assertion(
+    operator: &JsLogicalOperator,
+    left_binary_expression: &JsBinaryExpression,
+    right_binary_expression: &JsBinaryExpression,
+) -> bool {
+    let left_identifier = left_binary_expression.left().ok();
+    let right_identifier = right_binary_expression.right().ok();
+
+    if !matches!(operator, JsLogicalOperator::LogicalOr)
+        || !is_same_identifier(&left_identifier, &right_identifier)
+    {
+        return false;
+    }
+
+    let left_literal = extract_string_value(left_binary_expression.right().as_ref().ok());
+    let right_literal = extract_string_value(right_binary_expression.left().as_ref().ok());
+
+    match (left_literal, right_literal) {
+        (None, None) => false,
+        (Some(_), None) => true,
+        (None, Some(_)) => true,
+        (Some(left_value), Some(right_value)) => {
+            compare_literals(left_value.as_str(), right_value.as_str())
+        }
+    }
+}
+
+fn is_wrapped_in_parenthesis(logical_expression: &JsLogicalExpression) -> bool {
```

Comment:
The name of the function isn't aligned with the implementation. The implementation checks more than parenthesis. I suggest 
- rename the function 
- or document it

Reply:
I wasn't aware of `.prev_sibling_or_token()` and `.next_sibling_or_token()`, so I ended up with this hacky solution, but I just found out about it and now the implementation is more aligned with the name: it is actually looking for `(` and `)` tokens